docs/src/pages/docs/functions/helpers.md

```diff
@@ -280,8 +280,97 @@ Pass this function a function and it will return you a function that can be
 called in any form that you require until all arguments have been provided. For
 example if you pass a function: `f : (a, b, c) -> d` you get back a function
 that can be called in any combination, such as: `f(x, y, z)`, `f(x)(y)(z)`,
-`f(x, y)(z)`, or even `f(x)(y, z)`. This is great for doing partial application
-on functions for maximum re-usability.
+`f(x, y)(z)`, or even `f(x)(y, z)`. That is to say, this function fulfills the role
+of both curry and uncurry, returning a function that can be used as a curried
+function, an uncurried function, or any combination of argument applications in
+between. The returned function has `2^(n-1)` type signatures, where `n` is the
+number of parameters.
+
+```javascript
+import compose from 'crocks/helpers/compose'
+import curry from 'crocks/helpers/curry'
+import map from 'crocks/pointfree/map'
+import prop from 'crocks/maybe/prop'
+
+const crocksCurriedFunc =
+  curry((a, b, c, d) => a + b + c + d)
+
+crocksCurriedFunc(1)(2)(3)(4)  // [Fully curried]
+crocksCurriedFunc(1)(2)(3, 4)
+crocksCurriedFunc(1)(2, 3)(4)
+crocksCurriedFunc(1, 2)(3)(4)
+crocksCurriedFunc(1)(2, 3, 4)
+crocksCurriedFunc(1, 2, 3)(4)
+crocksCurriedFunc(1, 2, 3, 4)  // [Fully uncurried]
+crocksCurriedFunc(1, 2)(3, 4)
+
+// add :: (Number, Number, Number) -> Number
+const add = (a, b, c) =>
+  a + b + c
+
+// partial application impossible
+add(1)
+//=> NaN
+// 1 + undefined + undefined => NaN
+
+const curriedAdd =
```

Comment:
```javascript
// curriedAdd :: Number -> Number -> Number -> Number
```

Reply:
I can put that in. Despite what you said on the issue I raised, I am really resisting labeling function with a single type signature that actually has four. For the original definition of `curry` I understand the reasoning, but I'm making one last plea that it shouldn't be repeated too often when it isn't accurate. My feeling is that coder's in general, and functional coder's in particular, will assume the definition is perfectly correct and that they're looking at it wrong.

Reply:
I get that, mostly we are just matching the way we represent these handy auto-curry functions in JS. If you look at libs like Ramda, Fluture, Sanctuary (because of their ties to Ramda) they use this same signature to represent an auto-curried function.

I agree with you, other people coming from Haskell, Purescript, etc they may be a little unnerved by this. But people coming from FP from a JS side of things are already familiar with this short hand convention.

Reply:
I put it in. I really want to put a caveat on it, but I resisted the urge.

Reply:
So something we've been playing with the idea for a while now is having a section that's articles/tip & tricks/examples/blogs etc on the help file to guide people in starting out with crocks and potentially FP. Maybe when we have this section ready you could write something in more detail that really outlines everything in a way that shines the light on how to use it